src/components/engagement/engagement.service.ts

```diff
@@ -15,8 +26,274 @@ import {
+          canReadFirstScripture.canReadFirstScripture as canReadFirstScripture,
+          canEditFirstScripture.canEditFirstScripture as canEditFirstScripture,
+          canReadLukePartnership.canReadLukePartnership as canReadLukePartnership,
+          canEditLukePartnership.canEditLukePartnership as canEditLukePartnership,
+          canReadSentPrintingDate.canReadSentPrintingDate as canReadSentPrintingDate,
+          canEditSentPrintingDate.canEditSentPrintingDate as canEditSentPrintingDate,
+          canReadStatus.canReadStatus as canReadStatus,
+          canEditStatus.canEditStatus as canEditStatus,
+          canReadCeremony.canReadCeremony as canReadCeremony,
+          canEditCeremony.canEditCeremony as canEditCeremony,
+          canReadCompleteDate.canReadCompleteDate as canReadCompleteDate,
+          canEditCompleteDate.canEditCompleteDate as canEditCompleteDate,
+          CanReadDisbursementCompleteDate.CanReadDisbursementCompleteDate as CanReadDisbursementCompleteDate,
+          CanEditDisbursementCompleteDate.CanEditDisbursementCompleteDate as CanEditDisbursementCompleteDate,
+          canReadCommunicationsCompleteDate.canReadCommunicationsCompleteDate as canReadCommunicationsCompleteDate,
+          canEditCommunicationsCompleteDate.canEditCommunicationsCompleteDate as canEditCommunicationsCompleteDate,          
+          canReadModifiedAt.canReadModifiedAt as canReadModifiedAt,
+          canEditModifiedAt.canEditModifiedAt as canEditModifiedAt,
+          canReadStartDate.canReadStartDate as canReadStartDate,
+          canEditStartDate.canEditStartDate as canEditStartDate,
+          canReadEndDate.canReadEndDate as canReadEndDate,
+          canEditEndDate.canEditEndDate as canEditEndDate,
+          canReadInitialEndDate.canReadInitialEndDate as canReadInitialEndDate,
+          canEditInitialEndDate.canEditInitialEndDate as canEditInitialEndDate,
+          canReadLastSuspendedAt.canReadLastSuspendedAt as canReadLastSuspendedAt,
+          canEditLastSuspendedAt.canEditLastSuspendedAt as canEditLastSuspendedAt,
+          canReadLastReactivatedAt.canReadLastReactivatedAt as canReadLastReactivatedAt,
+          canEditLastReactivatedAt.canEditLastReactivatedAt as canEditLastReactivatedAt,
+          canReadStatusModifiedAt.canReadStatusModifiedAt as canReadStatusModifiedAt,
+          canEditStatusModifiedAt.canEditStatusModifiedAt as canEditStatusModifiedAt
+        `,
+        {
+          token: session.token,
+          requestingUserId: session.userId,
+          owningOrgId: session.owningOrgId,
+          id,
+        }
+      )
+      .first();
+
+    if (!result) {
+      throw new NotFoundException(
+        'Could not find langage engagement or internship engagement'
+      );
+    }
+
+    let language: Language = RedactedLanguage;
+    if (result.canReadLanguage) {
+      language = await this.languageService.readOne(result.languageId, session);
+    }
```

Comment:
If the user cannot read it, the value should be `undefined`. There's no need for `RedactedLanguage`.